main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -1315,12 +1315,18 @@ object Desugar {
 
   /**
     * Rewrites [[WeededAst.Expr.ListLit]] (`1 :: 2 :: Nil`) expression into `List.Nil` with `List.Cons`.
+    * If there are over 20 literals we translate it to `Vector.toList(Vector#{1, 2, ...})`.
```

Comment:
You are confusing two cases. THIS should ALWAYS be desugared.

The other case I am talking about is called FCons where you need a more tricky check.

Reply:
This case is `List#{1, 2, 3}` which is always desugared.